pkg/collector/corechecks/snmp/checkconfig/config_oid.go

```diff
@@ -0,0 +1,34 @@
+package checkconfig
+
+// OidConfig holds configs for OIDs to fetch
+type OidConfig struct {
+	// ScalarOids are all scalar oids to fetch
+	ScalarOids []string
+	// ColumnOids are all column oids to fetch
+	ColumnOids []string
+}
+
+func (oc *OidConfig) addScalarOids(oidsToAdd []string) {
+	oc.ScalarOids = oc.addOidsIfNotPresent(oc.ScalarOids, oidsToAdd)
+}
+
+func (oc *OidConfig) addColumnOids(oidsToAdd []string) {
+	oc.ColumnOids = oc.addOidsIfNotPresent(oc.ColumnOids, oidsToAdd)
+}
+
+func (oc *OidConfig) addOidsIfNotPresent(configOids []string, oidsToAdd []string) []string {
+	for _, oidToAdd := range oidsToAdd {
+		isAlreadyPresent := false
+		for _, oid := range configOids {
```

Comment:
There is no `set` in golang unfortunately.

It's possible to use a map instead.

In this use case, since we call addScalarOids/addColumnOids only few times, I think perf is not an issue.